plugins/groovy/test/org/jetbrains/plugins/groovy/lang/highlighting/GrResultOfAssignmentUsedTest.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.plugins.groovy.lang.highlighting
+
+import com.intellij.codeInspection.InspectionProfileEntry
+import org.jetbrains.plugins.groovy.codeInspection.assignment.GroovyResultOfAssignmentUsedInspection
+
+class GrResultOfAssignmentUsedTest extends GrHighlightingTestBase {
+  @Override
+  InspectionProfileEntry[] getCustomInspections() {
+    [new GroovyResultOfAssignmentUsedInspection()]
+  }
+
+  final warnStart = '<warning descr="Result of assignment expression used">'
+  final warnEnd = '</warning>'
+
+  void testUsedVar() {
+    testHighlighting """
+      def foo(a) {
+        if ((${warnStart}a = 5${warnEnd}) || a) {
+          a = 4
```

Comment:
since this is like a `def` setter method, I don't think this needs a warning (neither does the new inspection :p)